### ISSUE TITLE
tf.losses.log_losses calculation may be incorrect

### ISSUE THREAD

**Issue by piyank22** [tensorflow/tensorflow#12426]
https://github.com/tensorflow/tensorflow/blob/a0bbfa6afec617697576210fb22f3ea9c3a57b61/tensorflow/python/ops/losses/losses_impl.py#L409

here we are only considering one extreme that the value of prediction can be "zero", missing out probably that it can be "one" also.

If the variables predictions value is 1 then this would return a log loss of 
-log(1+1e-7) = -4.3429446044209946e-08
wherein if such occurs multiple times than it can indicate a lower loss.

I suggest clipping the value between [ 0.0000001 , 0.9999999 ]
